src/fragments/start/getting-started/flutter/integrate.mdx

```diff
@@ -228,230 +202,370 @@ class _AddTodoFormState extends State<AddTodoForm> {
+                      child: Text(_titleText),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
 }
 ```
 
-If you restart your app now, you should see that newly added todos will also start showing up on the list. The items look like they can be check off and marked as completed, but when pressed, they don’t seem to do anything right now. Let’s learn how to update existing data.
+Then, add the route to the global `GoRouter`.
 
-### Updating a Todo
-
-Updating an existing data entry looks a lot like creating a new one. It’s important to note, however, that models in DataStore are *immutable*. So, to update a record you must use a model’s `copyWith` function rather than manipulating its properties directly.
-
-Update the `_toggleIsComplete()` function in the `TodoItem` class.
-
-```dart
-Future<void> _toggleIsComplete() async {
+```diff
+  static final _router = GoRouter(
+    routes: [
+      GoRoute(
+        path: '/',
+        builder: (context, state) => const HomeScreen(),
+      ),
++     GoRoute(
++       path: '/manage-budget-entry',
++       name: 'manage',
++       builder: (context, state) => ManageBudgetEntryScreen(
++         budgetEntry: state.extra as BudgetEntry?,
++       ),
++     ),
+    ],
+  );
+```
 
-  // copy the Todo you wish to update, but with updated properties
-  final updatedTodo = todo.copyWith(isComplete: !todo.isComplete);
-  try {
+Finally, hook up the `+` button to navigate to this route when it's pressed.
 
-    // to update data in DataStore, you again pass an instance of a model to
-    // Amplify.DataStore.save()
-    await Amplify.DataStore.save(updatedTodo);
-  } catch (e) {
-    safePrint('An error occurred while saving Todo: $e');
+```diff
+  Future<void> _navigateToBudgetEntry({BudgetEntry? budgetEntry}) async {
+-    // To be filled in
++    await context.pushNamed('manage', extra: budgetEntry);
   }
-}
 ```
 
-That’s it! Restart your app and you should now be able to toggle a todo between completed and not completed states. You’re almost done here but what if you want to delete an item from your todo list? We’ll go over how to do that next.
+Now, if you click the `+` button you should be sent to the new screen.
+
+![Budget Tracker create screen](/images/lib/getting-started/flutter/budget-tracker-create-screen.png)
+
+Try creating a budget entry at this point! You should be able to enter values in the fields,
+and get errors if a value entered does not match the schema.
+
+When returning to the home screen, though, our budget entry is nowhere to be found! Let's fix that.
+
+### Loading remote budget entries
+
+In the `_HomeScreenState` class, we have a method called `_refreshBudgetEntries`. Let's populate
+that to update the state class's `_budgetEntries` variable every time we call it.
+
+To that, we'll use Amplify's so-called model helpers which provide methods we can call to quickly
```

Comment:
can we remove "so-called" here please?

Reply:
can we also simplify the language,

"Well use Amplify's model helpers, which provide a quick way to create CRUD requests for your data.